src/main/java/org/jitsi/jicofo/bridge/BridgeSelectionStrategy.java

```diff
@@ -86,61 +93,248 @@ public Bridge select(
     }
 
     /**
-     * Selects a bridge to be used for a new participant in a conference,
-     * assuming that no other bridge is used by the conference (i.e. this
-     * is the initial selection of a bridge for the conference).
+     * A1. (Happy case 1): There is a NOL bridge in the conference and in the
+     * region. Use the least loaded of them.
      *
-     * @param bridges the list of bridges to select from.
-     * @param participantRegion the region of the participant for which a
-     * bridge is to be selected.
-     * @return the selected bridge, or {@code null} if no bridge is
-     * available.
+     * @param bridges
+     * @param conferenceBridges
+     * @param participantRegion
+     *
+     * @return
      */
-    private Bridge selectInitial(List<Bridge> bridges,
-                                 String participantRegion)
+    private Optional<Bridge> a1(List<Bridge> bridges,
```

Comment:
I think it'd be clearer to try and come up with descriptive names for these helpers.  Like `findBridgeAlreadyInConferenceInRegion`.  Personally I'd rather have a long, descriptive name.  Then the cascade of calls ends up looking like:

```
findBridgeAlreadyInConferenceInRegion(bridges, conferenceBridges, participantRegion).orElse(
    findNewBridgeInRegion(bridges, conferenceBridges, participantRegion).orElse(
        ...
```

Which I think reads better.  (I think the fact that we're looking for a non-overloaded bridged in each case is implicit and could just be documented on the method).

Reply:
I restructured the code because the previous version was always evaluating the expression in the orElse() method call, even if it wasn't necessary.  So unfortunately now the method name findBridgeAlreadyInConferenceInRegionOrRestOfDecisionTree which is too long I think. Any suggestions how to deal with that?

Reply:
I brought back the previous code structure and tried to come up with better method names. I also removed rule b1 which identical to a2 code-wise.